src/index.ts

```diff
@@ -182,7 +182,8 @@ export default function() {
         if (!args.noCompile) {
           await bre.run("compile");
         }
-        return deploymentsManager.runDeploy(args.tags, {
+        await bre.run("deploy:watch:before", args);
```

Comment:
2 points:
- I would not call them `deploy:watch:*` as watch is not necessarely involved here
- I think only one internalTask is needed : `deploy:perform` or something

then you can override it and perform stuff before calling `runSuper` and then do things after it
the task would return the deploymentResult too so you can do stuff with it

Reply:
1. Agree on the naming. 
2. Are you suggesting to put compileAndDeploy function in an internal task that returns a promise of the deploymentResult? I guess that would be nice, but im not aquanited enugh with buidler-deploy code to know if this is a good idea.